pkg/utils/parse_arf_result.go

```diff
@@ -717,54 +717,89 @@ func toArrayByComma(format string) []string {
 func parseValues(remContent string, resultValues map[string]string) (string, []string, []string, error) {
 	var valuesUsedList []string
 	var valuesMissingList []string
-	var valuesParsedList []string
+
 	//find everything start and end with {{}}
 	re := regexp.MustCompile(`\{\{[^}]*\}\}`)
 	contentList := re.FindAllString(remContent, -1)
 	fixedText := remContent
 	if len(contentList) == 0 {
 		return remContent, valuesUsedList, valuesMissingList, nil
-	} // no processing needed, no urlencoded data
-	//process urlencoded content file data one by one and replace them in fixedText
+	}
+	// there are two types of content we need to process, one is url-encoded machine config source ex. {{ -a%20always0-F%20di }},
+	// the other one is not url_encoded ex. {{.var_some_value}}, we are going to take care of url-encoded content first, and then
+	// feed the processed content to template again
+
 	for _, content := range contentList {
-		//take out `{{ `,' }}' from content
+		// if third character is not space, skip non url-encoded content
+		if content[2:3] != " " {
```

Comment:
I still feel shaky about this, since it feels to me like a fragile check. Can we use a regex or something else instead?

Reply:
huh... isn't this already addressed in line 740? do we need this if statement?